### DISCUSSION TITLE
Support `--fix-dry-run` option in programmatic API

### DISCUSSION THREAD

**Discussion by jopemachine** [eslint/eslint#15877] [General] [answered]
`--fix-dry-run` option looks like it is available only through CLI.

I'm wondering if supporting this flag in programmatic API is possible.

I think it would be useful to use this option in eslint wrapper libraries like `xo` (Ref: https://github.com/xojs/xo/issues/663)

## References

- eslint's programmatic API
https://eslint.org/docs/developer-guide/nodejs-api#parameters

- eslint's CLI API
https://eslint.org/docs/user-guide/command-line-interface#--fix-dry-run

**Comment by mdjermanovic** [comment, marked as answer]
Can you elaborate more on what would be the functionality of this flag in ESLint API?

Linting and saving the changes are already two separate steps:

https://github.com/eslint/eslint/blob/ee69cd30b3551b3adebfd959a44a9a149221946a/lib/cli.js#L295-L307

* `eslint --fix` calls [`eslint.lintFiles`](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlintfilespatterns), and then [`ESLint.outputFixes`](https://eslint.org/docs/developer-guide/nodejs-api#-eslintoutputfixesresults).
* `eslint --fix-dry-run` only calls [`eslint.lintFiles`](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlintfilespatterns).

If you want to get the fixed versions (with the remaining unfixable problems) without modifying the files, you can just call `eslint.lintFiles`.

**Reply by jopemachine** [reply]
I didn't noticed CLI's `fixDryRun` option is [translated to `fix` option.](https://github.com/eslint/eslint/blob/main/lib/cli.js#L96)
I think I can just pass `fix: true` to ESLint's constructor option and call `eslint.lintFiles` to `fixDryRun`.
I appreciate your insightful answer.
